UnitedIncome/components
# [APP-2931] Box Component

### Description
> Please provide a brief description of your changes below.

Adds the `<Box />` component, which lets you display a label, description and icon in a convenient box. 

When an icon is provided, the `illuminate` prop is linked to the parent components `selected` prop, which means when the box is in a selected state the illustration will activate.

![image](https://user-images.githubusercontent.com/10888441/60197844-48cefe80-980e-11e9-90c4-6e5d08349366.png)

![image](https://user-images.githubusercontent.com/10888441/60197856-4f5d7600-980e-11e9-851a-7197d5811080.png)


### Testing Instructions
> What should be done in order to test this pull request, or re-create its effects.

* Load up the story in storybook and take a look.

### Comments
> Any additional comments that will aid with the review process.

This pull request does not include the BoxSelection component, that will fast follow. I wanted to seperate them out as they are separate tickets.

- [JIRA Ticket](https://unitedincome.atlassian.net/browse/app-2931)
